### ISSUE TITLE
macos系统初始化数据库时，报错了

### ISSUE THREAD

**Issue by linnengbo** [chatchat-space/Langchain-Chatchat#3035]
(base) apple@Mac Langchain-Chatchat % python init_database.py --recreate-vs
2024-02-22 10:19:14,731 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
recreating all vector stores
2024-02-22 10:19:15,366 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
2024-02-22 10:19:15,384 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
2024-02-22 10:19:15,385 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
2024-02-22 10:19:15,385 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
2024-02-22 10:19:15,390 - faiss_cache.py[line:94] - INFO: loading vector store in 'test/vector_store/bge-large-zh' from disk.
2024-02-22 10:19:15,390 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
2024-02-22 10:19:15,576 - SentenceTransformer.py[line:66] - INFO: Load pretrained SentenceTransformer: /Users/apple/IdeaProjects/models/BAAI/bge-large-zh
Batches: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  5.18it/s]
2024-02-22 10:19:16,812 - loader.py[line:64] - INFO: Loading faiss.
2024-02-22 10:19:16,908 - loader.py[line:66] - INFO: Successfully loaded faiss.
/Users/apple/miniconda3/lib/python3.11/site-packages/langchain_community/vectorstores/faiss.py:121: UserWarning: Normalizing L2 is not applicable for metric type: METRIC_INNER_PRODUCT
  warnings.warn(
zsh: segmentation fault  python init_database.py --recreate-vs
(base) apple@Mac Langchain-Chatchat % /Users/apple/miniconda3/lib/python3.11/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Comment by linnengbo** [comment]
> _🤖_
> 
> 嗨，@linnengbo！很高兴再次见到你。希望你一切都好。
> 
> 根据您提供的错误信息，这个问题可能与设备选择有关。在初始化数据库时，程序试图在一个不存在的设备上运行，这可能是因为您的系统没有正确配置CUDA或者您的机器上没有安装GPU。
> 
> 首先，您可以检查一下您的系统是否有可用的GPU设备。您可以通过运行以下Python代码来检查：
> 
> ```python
> import torch
> print(torch.cuda.device_count())
> ```
> 
> 如果返回的结果是0，那么说明您的系统上没有可用的GPU设备。在这种情况下，您需要在代码中指定使用CPU设备。如果返回的结果大于0，那么说明您的系统上有可用的GPU设备，您需要确保在代码中指定正确的设备编号。
> 
> 另外，根据我在Langchain-Chatchat仓库中找到的一些类似问题，这个问题可能也与`bge-large-zh`模型的加载有关。您可以尝试以下步骤来解决这个问题：
> 
> 1. 如果还没有安装`sentence-transformers`包，通过执行`pip install sentence-transformers`来安装。
> 2. 手动从Hugging Face下载`bge-large-zh`模型。
> 3. 修改`model_config.py`文件，更新`EMBEDDING_MODEL`和`MODEL_PATH`为模型存储的正确路径。
> 4. 修改完`model_config.py`后，运行`python copy_config_example.py`来应用更改。
> 
> 如果以上步骤不能解决您的问题，或者您需要进一步的帮助，请提供更多的信息，例如您的系统环境、Python版本、PyTorch版本等，这将有助于我们更好地理解和解决您的问题。
> 
> ## Sources
> * [[BUG] Langchain-chatchat 一键启动项目 python startup.py -a 报错](https://github.com/chatchat-space/Langchain-Chatchat/issues/2963)
> * [运行python init_database.py --recreate-vs  报错](https://github.com/chatchat-space/Langchain-Chatchat/issues/2096)
> * [[BUG] 3. 初始化知识库和配置文件 报错  python init_database.py --recreate-vs](https://github.com/chatchat-space/Langchain-Chatchat/issues/2283)
> * [init_database.py](https://github.com/chatchat-space/Langchain-Chatchat/blob/master/init_database.py)
> 
> #### About Dosu
> This response is meant to be useful and save you time. It is **not meant to be a precise solution**, but rather a starting point for your own research.
> 
> Help me be more useful! Please leave a 👍 if this is helpful and 👎 if it is irrelevant.
> 
> If you want to continue the conversation, start your reply with @dosu-bot.


是没有gpu的，用的是macos，但是我在llm_api_stale.py把默认的gpu改为cpu后，initdatabase还是一样的错，是有哪里还需要改吗？

**Comment by cszhangwenhua** [comment]
初步怀疑faiss未安装，
conda install faiss-cpu=1.7.4 -c pytorch
安装后，再执行，一直运行中，1小时多还未结束

**Comment by caijimhhhh** [comment]
faiss-cpu==1.7.4，仍然报错。

**Comment by caijimhhhh** [comment]
> 初步怀疑faiss未安装， conda install faiss-cpu=1.7.4 -c pytorch 安装后，再执行，一直运行中，1小时多还未结束

faiss版本正确，还是报错，请问还有什么额外的步骤吗？

**Comment by xiaogui340** [comment]
我也遇到这个问题了，求解决方案
配置：mac inter。 faiss-cpu=1.7.4
最新的项目版本
